File: KwetterWeb/kwetter/src/app/model/kweet/kweet.ts
export class Kweet {
  id: number;
  kweetText: string;
  userName: string;
  createdAt: Date;
  tags: string[];
  mentionedUsers: string[];

  constructor(id: number, kweetText: string, userName: string, createdAt: Date, tags: string[] | null, mentionedUsers: string[] | null) {
    this.id = id;
    this.kweetText = kweetText;
    this.userName = userName;
    this.createdAt = createdAt;
    this.tags = tags || [];
    this.mentionedUsers = mentionedUsers || [];
  }
}

File: Kweeter/Domain/Models/Kweet.cs
namespace Domain.Models;

public class Kweet
{
    public Guid KweetId { get; set; }

    public Guid UserId { get; set; }
    public string UserName { get; set; }
    public string KweetText { get; set; }
    public List<string> Tags { get; set; }
    public List<string> MentionedUsers { get; set; }
    public List<string> ContentLinks { get; set; }
    public DateTime CreatedAt { get; set; }
}